### PR TITLE
x11-wm/2bwm: new package

### PR DESCRIPTION
**Pull request by jakelyz**
A fast, floating window manager

Bug: https://bugs.gentoo.org/697904
Signed-off-by: Jake Lilly <lillyjsm@gmail.com>
Package-Manager: Portage-2.3.76, Repoman-2.3.16

### CODE REVIEW

File: x11-wm/2bwm/2bwm-0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		SRC_URI="https://github.com/venam/2bwm/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+		KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="A fast, floating window manager"
+HOMEPAGE="https://github.com/venam/2bwm"
+
+LICENSE="ISC"
+SLOT="0"
+IUSE=""
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-0.3.ebuild]: Drop it, if empty.

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+		SRC_URI="https://github.com/venam/2bwm/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+		KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="A fast, floating window manager"
+HOMEPAGE="https://github.com/venam/2bwm"
+
+LICENSE="ISC"
+SLOT="0"
+IUSE=""
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: Drop it, if empty.

File: x11-wm/2bwm/2bwm-0.3.ebuild (hunk 1 above)
Comment (jakelyz) [reply x11-wm/2bwm/2bwm-0.3.ebuild]: dropped

File: x11-wm/2bwm/2bwm-9999.ebuild (hunk 2 above)
Comment (jakelyz) [reply x11-wm/2bwm/2bwm-9999.ebuild]: dropped

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: Between EAPI and inherit should be an empty line.

File: x11-wm/2bwm/2bwm-0.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-0.3.ebuild]: Between EAPI and inherit should be an empty line.

File: x11-wm/2bwm/2bwm-9999.ebuild (hunk 3 above)
Comment (jakelyz) [reply x11-wm/2bwm/2bwm-9999.ebuild]: Added empty line

File: x11-wm/2bwm/2bwm-0.3.ebuild (hunk 4 above)
Comment (jakelyz) [reply x11-wm/2bwm/2bwm-0.3.ebuild]: Added empty line

**Comment by juippis** [comment]
@ConiKost ping, could you check again?

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+inherit savedconfig
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: Please use `toolchain-funcs`, in order to set a cc compiler.

Currently, it only calls `cc`
-> cc [..] -c -o 2bwm.o 2bwm.c

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+		KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="A fast, floating window manager"
+HOMEPAGE="https://github.com/venam/2bwm"
+
+LICENSE="ISC"
+SLOT="0"
+
+DEPEND="
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: It seems, that you are missing dependencies.

My compiled package uses different. Could you recheck on your side?

```
  NEEDED               libxcb.so.1 (x11-libs/libxcb)
  NEEDED               libxcb-randr.so.0 (x11-libs/libxcb)
  NEEDED               libxcb-keysyms.so.1 (x11-libs/xcb-util-keysyms)
  NEEDED               libxcb-icccm.so.4 (x11-libs/xcb-util-wm)
  NEEDED               libxcb-ewmh.so.2 (x11-libs/xcb-util-wm)
  NEEDED               libxcb-xrm.so.0 (x11-libs/xcb-util-xrm)
```

File: x11-wm/2bwm/2bwm-0.3.ebuild [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+		KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="A fast, floating window manager"
+HOMEPAGE="https://github.com/venam/2bwm"
+
+LICENSE="ISC"
+SLOT="0"
+
+DEPEND="
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-0.3.ebuild]: It seems, that you are missing dependencies.

My compiled package uses different. Could you recheck on your side?

```
  NEEDED               libxcb.so.1 (x11-libs/libxcb)
  NEEDED               libxcb-randr.so.0 (x11-libs/libxcb)
  NEEDED               libxcb-keysyms.so.1 (x11-libs/xcb-util-keysyms)
  NEEDED               libxcb-icccm.so.4 (x11-libs/xcb-util-wm)
  NEEDED               libxcb-ewmh.so.2 (x11-libs/xcb-util-wm)
  NEEDED               libxcb-xrm.so.0 (x11-libs/xcb-util-xrm)
```

File: x11-wm/2bwm/2bwm-0.3.ebuild [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit savedconfig
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-0.3.ebuild]: Please use `toolchain-funcs`, in order to set a cc compiler.

Currently, it only calls `cc`
-> cc [..] -c -o 2bwm.o 2bwm.c

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+if [[ ${PV} == 9999* ]] ; then
+		inherit git-r3
+		EGIT_REPO_URI="https://github.com/venam/2bwm.git"
+else
+		SRC_URI="https://github.com/venam/2bwm/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+		KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="A fast, floating window manager"
+HOMEPAGE="https://github.com/venam/2bwm"
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: https please.

File: x11-wm/2bwm/metadata.xml [hunk 10]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+<maintainer type="person">
```
Comment (ConiKost) [inline x11-wm/2bwm/metadata.xml]: Please use intends :-)

File: x11-wm/2bwm/2bwm-0.3.ebuild (hunk 7 above)
Comment (jakelyz) [reply x11-wm/2bwm/2bwm-0.3.ebuild]: dropped xcb-util in favor of libxcb. This appeared to be okay.

File: x11-wm/2bwm/2bwm-0.3.ebuild [hunk 11]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit savedconfig toolchain-funcs
+
+if [[ ${PV} == 9999* ]] ; then
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-0.3.ebuild]: `if [[ ${PV} == *9999 ]]; then`

File: x11-wm/2bwm/2bwm-9999.ebuild [hunk 12]
```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit savedconfig toolchain-funcs
+
+if [[ ${PV} == 9999* ]] ; then
```
Comment (ConiKost) [inline x11-wm/2bwm/2bwm-9999.ebuild]: `if [[ ${PV} == *9999 ]]; then`

**Pull request closed without merging**